AccountingSystem/Agent/NetworkAgent.py

```diff
@@ -161,47 +161,47 @@ def processMessage( self, headers, body ):
       metadataKey += value
 
     timestamps = sorted( body['datapoints'] )
-    
-    # approximate the value of first the timestamp
-    # (datapoints are take every 1 minute)
-    previousTimestamp = float( timestamps[0] ) - 60
-    for nextTimestamp in timestamps:
-    
-      data = body['datapoints'][nextTimestamp]
-
+    for timestamp in timestamps:
       try:
-        # set the time and create a hash
-        startTime = datetime.utcfromtimestamp( previousTimestamp )
-        endTime = datetime.utcfromtimestamp( float( nextTimestamp ) )
-
-        key = "%s%s%s" % ( metadataKey, str( startTime ), str( endTime ) )
+        date = datetime.utcfromtimestamp( float( timestamp ) )
+        
+        # create a key that allows to join packet-loss-rate and one-way-delay
+        # metrics in one network accounting record 
+        networkAccountingObjectKey = "%s%s" % ( metadataKey, str( date ) )
 
         # use existing or create a new temporary accounting
         # object to store the data in DB
-        if self.buffer.has_key( key ):
-          net = self.buffer[key]['object']
+        if self.buffer.has_key( networkAccountingObjectKey ):
+          net = self.buffer[networkAccountingObjectKey]['object']
+          
+          timeDifference = datetime.now() - self.buffer[networkAccountingObjectKey]['addTime']
+          if timeDifference.total_seconds() > 60:
+            self.log.warn( 'Object was taken from buffer after %s' % ( timeDifference ) )
         else:
           net = Network()
-          net.setStartTime( startTime )
-          net.setEndTime( endTime )
+          net.setStartTime( date )
+          net.setEndTime( date )
```

Comment:
why now they are the same?

Reply:
The packet-loss-rate and one-way-delay data send in raw perfSONARS streams are just data points taken at some given time (comparing to summaries which were averages over some time span). Those data points are taken around every 60s (around, because sometime they are taken later when perfSONAR host was busy with other things).

At the beginning I though it is important to have time spans (as previously when we had 5 minutes summaries and they could span across two buckets), but I have just rethought this and after additional checks it seems that it is perfectly fine for the Accounting to put just time points rather than 60s time spans. This also removes approximation of the first element which I had introduced previously (done by previousTimestamp and nextTimestamp variables).